C-Plus-Plus/cp/Space_Array.cpp

```diff
@@ -0,0 +1,97 @@
+/*
+The rules of the game are as follows:
+
+1-Initially, a sequence a1,a2,…,aN is given.
+2-The players alternate turns.
+3-In each turn, the current player must choose
+an index i and increment ai, i.e. change ai to ai+1.
+4-Afterwards, if there is no permutation p1,p2,…,pN of the
+integers 1 through N such that ai=pi holds for each valid
+ i, the current player loses.
+5-Otherwise, the game continues with the next turn.
+*/
+
+/*
+Input:
+The first line of the input contains a single integer T denoting the number of test cases.
+The description of T test cases follows.
+The first line of each test case contains a single integer N.
+The second line contains N space-separated integers a1,a2,…,aN.
+Output:
+For each test case, print a single line containing the string "First"
+if the first player wins or "Second"
+if the second player wins (without quotes).*/
+
+/*
+Constraints
+1=T=2·104
+1=N=2·105
+The sum of N over all test cases doesn't exceed 2·105
+1=ai=N for each valid i
+*/
+
+// Code
+#include<bits/stdc++.h>
+using namespace std;
+int main() {
```

Comment:
Use a different function for the main logic of the code. Use `main()` function for the driver code